src/plugins/application_config/server/types.ts

```diff
@@ -0,0 +1,9 @@
+export interface ApplicationConfigPluginSetup {
+  setConfigurationClient: (inputConfigurationClient: ConfigurationClient) => void;
+}
+// eslint-disable-next-line @typescript-eslint/no-empty-interface
+export interface ApplicationConfigPluginStart {}
+
+export interface ConfigurationClient {
```

Comment:
could we add jsdoc to explain what does this interface expect?

Reply:
Sure. Will do.

Reply:
Added